### PR TITLE
release command

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Towards: [giantswarm/giantswarm#5206](https://github.com/giantswarm/giantswarm/issues/5206)

Provide a new `architect release` command which does the following:
- template the chart
- package it as an archive
- create a draft github release with the chart uploaded as asset

### CODE REVIEW

File: release/release.go [hunk 1]
```diff
@@ -12,38 +12,29 @@ import (
 	"github.com/giantswarm/microerror"
 )
 
-type releaseInfo struct {
```
Comment (TheoBrigitte) [inline release/release.go]: I removed `releaseInfo` here as it was just a lame sub copy of `ReleaseGithubTask`.

File: cmd/release.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+		Sha: sha,
+		Tag: tag,
+	}
+
+	var githubClient *github.Client
+	{
+		if deploymentEventsToken == "" {
+			log.Printf("no github token")
+			return
```
Comment (kopiczko) [inline cmd/release.go]: I think it should `os.Exit(1)` or something.

File: cmd/release.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+		}
+
+		ctx := context.Background()
```
Comment (kopiczko) [inline cmd/release.go]: Can we move this line to the top of the function to feed my ocd?

File: workflow/workflow.go [hunk 4]
```diff
@@ -276,6 +280,46 @@ func NewPublish(projectInfo ProjectInfo, fs afero.Fs) (Workflow, error) {
 	return w, nil
 }
 
+func NewRelease(projectInfo ProjectInfo, fs afero.Fs, githubClient *github.Client) (Workflow, error) {
+	w := Workflow{}
+
+	chartDirectory := filepath.Join(projectInfo.WorkingDirectory, "helm", projectInfo.Project+"-chart")
```
Comment (kopiczko) [inline workflow/workflow.go]: Now we support multiple chart tarballs per repo. This is useful for testing. How are we going to deal with that?

File: cmd/release.go (hunk 2 above)
Comment (TheoBrigitte) [reply cmd/release.go]: good point, replaced this with `log.Fatalf`

File: cmd/release.go (hunk 3 above)
Comment (TheoBrigitte) [reply cmd/release.go]: anything to feed your ocd sir

File: workflow/workflow.go (hunk 4 above)
Comment (TheoBrigitte) [reply workflow/workflow.go]: I am not sure what's the issue here, can you give an example please ?

Comment (kopiczko) [reply workflow/workflow.go]: One example is apiextensions repo: https://github.com/giantswarm/apiextensions/tree/master/helm

Comment (TheoBrigitte) [reply workflow/workflow.go]: hum right, then I'll have to rework this along the line of the `processHelmDir` function

**Comment by TheoBrigitte** [comment]
closing this as there was various changes, mainly support for multiple charts.
Other PR are here:
- https://github.com/giantswarm/architect/pull/289
- https://github.com/giantswarm/architect/pull/290

**Pull request closed without merging**